tracer/src/Datadog.Trace/DataStreamsMonitoring/VarEncodingHelper.cs

```diff
@@ -0,0 +1,477 @@
+    /// <param name="bytesRead">The number of bytes read when successfully decoded</param>
+    /// <returns>The decoded value, or null if decoding failed</returns>
+    public static long? ReadVarLongZigZag(byte[] bytes, int offset, out int bytesRead)
+        => ReadVarLong(bytes, offset, out bytesRead) is { } decoded ? UnZigZag(decoded) : null;
+
+#if NETCOREAPP3_1_OR_GREATER
+    /// <summary>
+    /// Deserializes 64-bit unsigned integers that have been encoded using
+    /// <see cref="WriteVarLongZigZag(byte[],int,long)"/>
+    /// </summary>
+    /// <param name="bytes">The buffer to read the encoded value from</param>
+    /// <param name="bytesRead">The number of bytes read when successfully decoded</param>
+    /// <returns>The decoded value, or null if decoding failed</returns>
+    public static long? ReadVarLongZigZag(Span<byte> bytes, out int bytesRead)
+        => ReadVarLong(bytes, out bytesRead) is { } decoded ? UnZigZag(decoded) : null;
+#endif
+
+    /// <summary>
+    /// Deserializes 64-bit unsigned integers that have been encoded using
+    /// <see cref="WriteVarInt(byte[],int,uint)" />
+    /// </summary>
+    /// <param name="bytes">The byte array to write the buffer into</param>
+    /// <param name="offset">The offset within the array to start writing</param>
+    /// <param name="bytesRead">The number of bytes read when successfully decoded</param>
+    /// <returns>The decoded value, or null if decoding failed</returns>
+    public static uint? ReadVarInt(byte[] bytes, int offset, out int bytesRead)
+        => ReadVarLong(bytes, offset, out bytesRead) is { } decoded and <= int.MaxValue
+            ? (uint)decoded
+            : null;
+
+#if NETCOREAPP3_1_OR_GREATER
+    /// <summary>
+    /// Deserializes 64-bit unsigned integers that have been encoded using
+    /// <see cref="WriteVarInt(byte[],int,uint)" />
+    /// </summary>
+    /// <param name="bytes">The buffer to read the encoded value from</param>
+    /// <param name="bytesRead">The number of bytes read when successfully decoded</param>
+    /// <returns>The decoded value, or null if decoding failed</returns>
+    public static uint? ReadVarInt(Span<byte> bytes, out int bytesRead)
+        => ReadVarLong(bytes, out bytesRead) is { } decoded and <= int.MaxValue
+            ? (uint)decoded
+            : null;
+#endif
+
+    /// <summary>
+    /// Deserializes 64-bit unsigned integers that have been encoded using
+    /// <see cref="WriteVarIntZigZag(byte[],int,int)" />
+    /// </summary>
+    /// <param name="bytes">The byte array to write the buffer into</param>
+    /// <param name="offset">The offset within the array to start writing</param>
+    /// <param name="bytesRead">The number of bytes read when successfully decoded</param>
+    /// <returns>The decoded value, or null if decoding failed</returns>
+    public static long? ReadVarIntZigZag(byte[] bytes, int offset, out int bytesRead)
```

Comment:
Is this supposed to be a `long`? Seems like it should be an `int` if it gets casted to an `int`, unsure though if I'm missing something about this though.

Same question as the `Span` version of this as well.

Reply:
Yep, you're right, good 👀, thanks!